Tests/NIOHTTP2Tests/ContentLengthVerifierTests.swift

```diff
@@ -0,0 +1,27 @@
+import NIOHPACK
+@testable import NIOHTTP2
+import XCTest
+
+class ContentLengthVerifierTests: XCTestCase {
+    func testDuplicatedLengthHeadersPermitted() throws {
+        var headers = HPACKHeaders([("Host", "apple.com"), ("content-length", "1834"), ("User-Agent", "myCoolClient/1.0")])
+        XCTAssertNoThrow(try ContentLengthVerifier(headers))
+
+        headers.add(contentsOf: [("content-length", "1834")])
+        XCTAssertNoThrow(try ContentLengthVerifier(headers))
+
+        headers.add(contentsOf: [("content-length", "1834")])
+        XCTAssertNoThrow(try ContentLengthVerifier(headers))
+
+        headers.add(contentsOf: [("Content-Length", "1834")])
+        XCTAssertNoThrow(try ContentLengthVerifier(headers))
+    }
+
+    func testDuplicatedConflictingLengthHeadersThrow() throws {
```

Comment:
Can you add a test for the case you've indicated in your PR comment, where the two numerical values match but the string values are different?